internal/transport/vim.go

```diff
@@ -0,0 +1,252 @@
+package transport
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/myitcv/govim/internal/queue"
+	"gopkg.in/tomb.v2"
+	"io"
+	"strings"
+	"sync"
+	"time"
+)
+
+type vimTransport struct {
+	in  *json.Decoder
+	out *json.Encoder
+	log io.Writer
+
+	// callVimNextID represents the next ID to use in a call to the Vim
+	// channel handler. This then allows us to direct the response.
+	callVimNextID     int
+	callbackResps     map[int]Callback
+	callbackRespsLock sync.Mutex
+
+	eventQueue *queue.Queue
+	tomb       tomb.Tomb
+}
+
+func NewVimTransport(in io.Reader, out io.Writer, log io.Writer, eventQueue *queue.Queue) Transport {
+	return &vimTransport{
+		in:  json.NewDecoder(in),
+		out: json.NewEncoder(out),
+		log: log,
+
+		callVimNextID: 1,
+		callbackResps: make(map[int]Callback),
+
+		eventQueue: eventQueue,
+	}
+}
+
+func (v *vimTransport) Start() error {
+	v.tomb.Go(v.run)
+	return nil
+}
+
+func (v *vimTransport) run() error {
+	for {
+		select {
+		case <-v.tomb.Dying():
+			return tomb.ErrDying
+		}
+	}
```

Comment:
At the moment this is purely just to keep the `tomb` alive, but I wonder if there's a better way of doing this. We might need this in future if we want to move more of the event loop into here